forge/src/main/java/org/openjdk/skara/forge/PullRequestPoller.java

```diff
@@ -0,0 +1,326 @@
+        quarantineMap.put(pr.id(), new PullRequestRetry(pr, until));
+    }
+
+    /**
+     * Queries the repository for pull requests. On the first round (or until any
+     * results have been received), get all pull requests. After that limit the
+     * results using the maxUpdatedAt value from the previous results. Use padding
+     * if needed to guarantee that we never miss an update.
+     */
+    private List<PullRequest> queryPullRequests() {
+        if (prev == null || prev.maxUpdatedAt == null) {
+            if (includeClosed) {
+                log.fine("Fetching all open and recent closed pull requests for " + repository.name());
+                // We need to guarantee that all open PRs are always included in the first round.
+                // The pullRequests(ZonedDateTime) call has a size limit, so may leave some out.
+                // There may also be open PRs that haven't been updated since the closed age limit.
+                var openPrs = repository.openPullRequests();
+                var allPrs = repository.pullRequestsAfter(ZonedDateTime.now().minus(CLOSED_PR_AGE_LIMIT));
+                return Stream.concat(openPrs.stream(), allPrs.stream().filter(pr -> !pr.isOpen())).toList();
+            } else {
+                log.fine("Fetching all open pull requests for " + repository.name());
+                return repository.openPullRequests();
+            }
+        } else {
+            var queryUpdatedAt = paddingNeeded ? prev.maxUpdatedAt.minus(queryPadding) : prev.maxUpdatedAt;
+            if (includeClosed) {
+                log.fine("Fetching open and closed pull requests updated after " + queryUpdatedAt + " for " + repository.name());
+                return repository.pullRequestsAfter(queryUpdatedAt);
+            } else {
+                log.fine("Fetching open pull requests updated after " + queryUpdatedAt + " for " + repository.name());
+                return repository.openPullRequestsAfter(queryUpdatedAt);
+            }
+        }
+    }
+
+    private Map<String, List<Comment>> fetchComments(List<PullRequest> prs, ZonedDateTime maxUpdatedAt) {
+        if (checkComments) {
+            return prs.stream()
+                    .filter(pr -> pr.updatedAt().isAfter(maxUpdatedAt.minus(queryPadding)))
+                    .collect(Collectors.toMap(Issue::id, Issue::comments));
+        } else {
+            return Map.of();
+        }
+    }
+
+    private Map<String, List<Review>> fetchReviews(List<PullRequest> prs, ZonedDateTime maxUpdatedAt) {
+        Map<String, List<Review>> reviewsMap;
+        if (checkReviews) {
+            reviewsMap = prs.stream()
+                    .filter(pr -> pr.updatedAt().isAfter(maxUpdatedAt.minus(queryPadding)))
+                    .collect(Collectors.toMap(Issue::id, PullRequest::reviews));
+        } else {
+            reviewsMap = Map.of();
+        }
+        return reviewsMap;
```

Comment:
Just an aesthetic remark. It looks a bit odd that this method has a local variable instead of returning directly, when it is almost the very same as the method above.

Reply:
Oh, I see what you mean now. Will fix.